angr/analyses/cfg/cfg_fast.py

```diff
@@ -1675,6 +1676,17 @@ def _func_addrs_from_prologues(self):
         for ins_regex in self.project.arch.function_prologs:
             r = re.compile(ins_regex)
             regexes.append(r)
+        # EDG says: I challenge anyone bothering to read this to come up with a better
+        # way to handle CPU modes that affect instruction decoding.
+        # Since the only one we care about is ARM/Thumb right now
+        # we have this gross hack. Sorry about that.
```

Comment:
realistically the answer is that we represent thumb mode with a separate arch but that's a nightmare for angr 9

Reply:
Currently we have the assumption that a project has only one `arch`... I tried to get rid of it when dealing with JNI support, and I gave up because it was such a nightmare.